test_pvc_expansion_when_full failing with error "No space left on device"

The test case test_pvc_expansion_when_full is failing with the error given below.

```
log_step(
        "Run IO on PVCs to utilize 91% of available storage. Check PersistentVolumeUsageNearFull alert."
    )

  self._run_io_and_check_alerts(
        fill_up_near_full_mb,
        prometheus_api,
        alert_type="PersistentVolumeUsageNearFull",
        alert_msg="is nearing full. Data deletion or PVC expansion is required.",
        threshold_attr="near_full_alert",
    )

tests/functional/pv/pvc_resize/test_pvc_expansion_when_full.py:86: 


..........


raise CommandFailed(
                f"Error during execution of command: {masked_cmd}."
                f"\nError is {masked_stderr}"
            )
E               ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc -n namespace-test-bc61098a0fb141538b7d08540 rsh pod-test-rbd-a733978e368548cbb0c3fd77722 fio --name=fio-rand-write --filename=/var/lib/www/html/pod-test-rbd-a733978e368548cbb0c3fd77722_f1 --rw=randwrite --bs=4K --direct=0 --numjobs=1 --time_based=1 --runtime=30 --size=3727M --ioengine=libaio --iodepth=4 --rate=100M --rate_process=poisson --output-format=json.
E               Error is fio: io_u error on file /var/lib/www/html/pod-test-rbd-a733978e368548cbb0c3fd77722_f1: No space left on device: write offset=2893082624, buflen=4096
E               fio: io_u error on file /var/lib/www/html/pod-test-rbd-a733978e368548cbb0c3fd77722_f1: No space left on device: write offset=3295195136, buflen=4096
E               fio: io_u error on file /var/lib/www/html/pod-test-rbd-a733978e368548cbb0c3fd77722_f1: No space left on device: write offset=1836834816, buflen=4096
E               fio: io_u error on file /var/lib/www/html/pod-test-rbd-a733978e368548cbb0c3fd77722_f1: No space left on device: write offset=1570689024, buflen=4096
E               command terminated with exit code 1



ocs_ci/utility/utils.py:709: CommandFailed
```

This issue was seen in 4.16.
Failed test runs - 
https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#ocs/launches/678/26219/1283545/1283589/log
https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#ocs/launches/678/26158/1280285/1280329/log
https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#ocs/launches/678/26144/1279470/1279514/log

This issue started after a recent change in the test case.

Test case - tests/functional/pv/pvc_resize/test_pvc_expansion_when_full.py::TestPvcExpansionWhenFull::test_pvc_expansion_when_full

--- a/tests/functional/pv/pvc_resize/test_pvc_expansion_when_full.py
+++ b/tests/functional/pv/pvc_resize/test_pvc_expansion_when_full.py
@@ -292,6 +292,7 @@ class TestPvcExpansionWhenFull(ManageTest):
                 io_direction="write",
                 runtime=30,
                 rate="100M",
+                fio_filename=f"file_{fill_up_mb}mb",
                 end_fsync=1,
             )
         for pod_obj in self.pods:

--- a/tests/functional/pv/pvc_resize/test_pvc_expansion_when_full.py
+++ b/tests/functional/pv/pvc_resize/test_pvc_expansion_when_full.py
@@ -105,7 +105,7 @@ class TestPvcExpansionWhenFull(ManageTest):
         log_step("Run IO on all to utilize 100% of PVCs storage capacity.")
         self.fill_up_pvcs(fill_up_full_mb, pvc_full_error_expected=True)
 
-        log_step("Run IO on PVCs to utilise 100% of PVCs storage capacity.")
+        log_step("Verify used space on pods is 100%.")
         self._verify_used_space_on_pods("100%")
 
         log_step("Expanding PVCs.")
@@ -120,12 +120,7 @@ class TestPvcExpansionWhenFull(ManageTest):
         self.verify_no_pvc_alerts(prometheus_api, round(timeout_alerts / 2))
 
         log_step("Run IO after PVC expansion.")
-        self.fill_up_pvcs(
-            fill_up_near_full_mb
-            + fill_up_critical_full_mb
-            + fill_up_full_mb
-            + (pvc_fill_up_after_resize * 1024)
-        )
+        self.fill_up_pvcs(pvc_fill_up_after_resize * 1024)
 
         log_step("Verify no PVC full alerts after PVC expansion and additional IO.")
         self.verify_no_pvc_alerts(prometheus_api, round(timeout_alerts / 2))
@@ -279,7 +274,7 @@ class TestPvcExpansionWhenFull(ManageTest):
 
     def fill_up_pvcs(self, fill_up_mb, pvc_full_error_expected=False):
         """
-        Run IO on pods to utilize the PVCs until near_full threshold
+        Run IO on pods to utilize the PVCs
 
         Args:
             fill_up_mb (int): The amount of data to write to the PVCs

--- a/tests/functional/pv/pvc_resize/test_pvc_expansion_when_full.py
+++ b/tests/functional/pv/pvc_resize/test_pvc_expansion_when_full.py
@@ -95,7 +95,7 @@ class TestPvcExpansionWhenFull(ManageTest):
             "Run IO on PVCs to utilize 96% of available storage. Check PersistentVolumeUsageCritical alert."
         )
         self._run_io_and_check_alerts(
-            fill_up_near_full_mb + fill_up_critical_full_mb,
+            fill_up_critical_full_mb,
             prometheus_api,
             alert_type="PersistentVolumeUsageCritical",
             alert_msg="is critically full. Data deletion or PVC expansion is required.",
@@ -103,10 +103,7 @@ class TestPvcExpansionWhenFull(ManageTest):
         )
 
         log_step("Run IO on all to utilize 100% of PVCs storage capacity.")
-        self.fill_up_pvcs(
-            fill_up_near_full_mb + fill_up_critical_full_mb + fill_up_full_mb,
-            pvc_full_error_expected=True,
-        )
+        self.fill_up_pvcs(fill_up_full_mb, pvc_full_error_expected=True)
 
         log_step("Run IO on PVCs to utilise 100% of PVCs storage capacity.")
         self._verify_used_space_on_pods("100%")
@@ -295,7 +292,7 @@ class TestPvcExpansionWhenFull(ManageTest):
                 io_direction="write",
                 runtime=30,
                 rate="100M",
-                fio_filename=f"{pod_obj.name}_f1",
+                end_fsync=1,
             )
         for pod_obj in self.pods:
             try:

